wasm/parsers/module.py

```diff
@@ -0,0 +1,66 @@
+import io
+
+from wasm.datatypes import (
+    Function,
+    Module,
+)
+from wasm.exceptions import (
+    MalformedModule,
+)
+
+from .magic import (
+    parse_magic,
+)
+from .sections import (
+    parse_sections,
+)
+from .version import (
+    parse_version,
+)
+
+
+def parse_module(stream: io.BytesIO) -> Module:
+    parse_magic(stream)
+    parse_version(stream)
```

Comment:
:+1: I think I'll probably go with docstring/comments to explain why `magic` is discarded and will add an extra field to the module to store the version so that we can apply a `validate_version` to it during validation.